wrapper/src/test/java/integration/refactored/container/tests/HikariTests.java

```diff
@@ -111,4 +142,185 @@ public void testOpenConnectionWithDataSourceClassName() throws SQLException {
+      ProxyHelper.disableConnectivity(writerIdentifier);
+
+      assertThrows(FailoverSuccessSQLException.class, () -> auroraUtil.queryInstanceId(conn));
+
+      // Check the connection is valid after connecting to a different instance
+      assertTrue(conn.isValid(5));
+      currentConnectionId = auroraUtil.queryInstanceId(conn);
+      LOGGER.fine("Connected to instance: " + currentConnectionId);
+      assertTrue(currentConnectionId.equalsIgnoreCase(readerIdentifier));
+    }
+  }
+
+  private void createDataSource(final Properties customProps) {
+    final HikariConfig config = getConfig(customProps);
+    dataSource = new HikariDataSource(config);
+
+    final HikariPoolMXBean hikariPoolMXBean = dataSource.getHikariPoolMXBean();
+
+    LOGGER.fine("Starting idle connections: " + hikariPoolMXBean.getIdleConnections());
+    LOGGER.fine("Starting active connections: " + hikariPoolMXBean.getActiveConnections());
+    LOGGER.fine("Starting total connections: " + hikariPoolMXBean.getTotalConnections());
+  }
+
+  private HikariConfig getConfig(Properties customProps) {
+    final HikariConfig config = new HikariConfig();
+    config.setUsername(TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getUsername());
+    config.setPassword(TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getPassword());
+    config.setMaximumPoolSize(3);
+    config.setReadOnly(true);
+    config.setExceptionOverrideClassName(HikariCPSQLException.class.getName());
+    config.setInitializationFailTimeout(75000);
+    config.setConnectionTimeout(1000);
+
+    config.setDataSourceClassName(AwsWrapperDataSource.class.getName());
+    config.addDataSourceProperty("targetDataSourceClassName", DriverHelper.getDataSourceClassname());
+    config.addDataSourceProperty("jdbcProtocol", DriverHelper.getDriverProtocol());
+    config.addDataSourceProperty("portPropertyName", "portNumber");
+    config.addDataSourceProperty("serverPropertyName", "serverName");
+    config.addDataSourceProperty("databasePropertyName", "databaseName");
+
+    Properties targetDataSourceProps = new Properties();
+
+    targetDataSourceProps.setProperty(
+        "serverName",
+        TestEnvironment.getCurrent()
+            .getInfo()
+            .getProxyDatabaseInfo()
+            .getInstances()
+            .get(0)
+            .getEndpoint());
+    targetDataSourceProps.setProperty(
+        "databaseName",
+        TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getDefaultDbName());
+
+    targetDataSourceProps.setProperty("portNumber",
+        Integer.toString(TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getClusterEndpointPort()));
+    targetDataSourceProps.setProperty("socketTimeout", "3");
+    targetDataSourceProps.setProperty("connectTimeout", "3");
+    targetDataSourceProps.setProperty("monitoring-connectTimeout", "1");
+    targetDataSourceProps.setProperty("monitoring-socketTimeout", "1");
```

Comment:
These parameters are driver specific. You need to use values in seconds or millis depending on a driver. It should be something like this:

            DriverHelper.setMonitoringConnectTimeout(props, CONNECT_TIMEOUT_SEC, TimeUnit.SECONDS);
            DriverHelper.setMonitoringSocketTimeout(props, TIMEOUT_SEC, TimeUnit.SECONDS);
            DriverHelper.setConnectTimeout(props, CONNECT_TIMEOUT_SEC, TimeUnit.SECONDS);